up-and-running-with-redux-observable/notes/03-redux-observable-add-redux-observable-to-an-existing-redux-project.md

```diff
@@ -0,0 +1,99 @@
+## 03. Add redux-observable to an existing Redux project
+
+<Timestamp start="0:30" end="1:30">
+    
+As of redux-observable 1.0.0, we no longer provide rootEpic to createEpicMiddleware. Instead, we call `epicMiddleware.run(rootEpic)` after creating our store with it. Our middleware setup should look as follows:
+
+```
+import { applyMiddleware, createStore } from "redux"
+import { createEpicMiddleware } from "redux-observable"
+import { rootEpic } from "./epics"
+
+const epicMiddleware = createEpicMiddleware();
+const store = createStore(reducer, applyMiddleware(epicMiddleware));
+epicMiddleware.run(rootEpic);
+```
+
+See https://redux-observable.js.org/MIGRATION.html for more information.
+
+</Timestamp>
+
+<Timestamp start="1:40" end="4:50">
+
+RxJS Operators are now piped to Observables using the `Observable.pipe(operator1, operator2, ...)` syntax. This means that we need to import each RxJS operator separately from `rxjs/operators`. Also, with pipeable operators, `do()` is replaced with `tap()` (see https://www.learnrxjs.io/learn-rxjs/operators/utility/do). Thus, to match the functionality demonstrated in the lesson our `epics/index.js` should look like
+
+```
+import { tap, ignoreElements, filter } from "rxjs/operators"
+import { combineEpics } from "redux-observable"
+import { LOAD_STORIES } from "../actions"
+
+function loadStoriesEpic(action$) {
+    return action$.pipe(
+        filter((action) => action.type === LOAD_STORIES),
+        tap((action) => console.log(action)),
+        ignoreElements()
+    )
+}
+
+export const rootEpic = combineEpics(loadStoriesEpic);
+```
+
+See https://rxjs.dev/guide/operators for more information.
```

Comment:
I think this is too long to fit in one note. Maybe separating them out into two different notes and shortening down the time would be better.